ide/lib.terminalemulator/src/org/netbeans/lib/terminalemulator/MyFontMetrics.java

```diff
@@ -142,18 +142,22 @@ public MyFontMetrics(Component component, Font font) {
         height -= leading;
         leading = 0;
 
-        cwidth_cache = CacheFactory.cacheForFontMetrics(fm);
+        cwidth_cache = CacheFactory.cacheForFontMetrics(getFm());
+    }
+
+    public FontMetrics getFm() {
+        return this.cmp.getFontMetrics(cmp.getFont());
```

Comment:
Is it necessary to store `cmp` in order to obtain `getFont` - couldn't be the font obtained / stored rather than the Component ? Not a big deal, but FontMetrics > Component seem as a backward dependency.

Reply:
@hlavki anything on this?

Reply:
@lkishalmi, sorry for delay. I didn't have time to fix @sdedic's suggestion. Maybe in next round. thanks

Reply:
@sdedic this is very interesting think, because in previous implementation `FontMetrics` was created in `MyFontMetrics` constructor, but it didn't work. Cursor was shifted. Would be great if somebody with strong swing knowledge could explain this behaviour.